ath79: add basic kernel 6.1 support

DragonBluep opened the pull request:
“Old, but not obsolete.”

Known issues:
- [x] ath10k-ct doesn't work.
It's a pcie init issue, fixed. Thanks to @robimarko
- [ ] sysupgrade warning `VFS: Busy inodes after unmount of jffs2`
This issue exists on all targets. Unrelated to ath79.
Ref: https://github.com/openwrt/openwrt/issues/12891
- [x] when downgrade from the 6.1 kernel via LuCI, sometimes device won't restart
Re-plug power supply can recover the system.
Can not reproduce it now. 2023/08/03
<details>
<summary>NETGEAR R6100 syslog:</summary>

```
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.000000] Linux version 6.1.29 (db@Aspire-V-Nitro) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 12.3.0 r23122-d39d4e7b84) 12.3.0, GNU ld (GNU Binutils) 2.40.0) #0 Tue May 30 03:59:39 2023
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] printk: bootconsole [early0] enabled
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] CPU0 revision is: 0001974c (MIPS 74Kc)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] MIPS: machine is Netgear R6100
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] SoC: Atheros AR9344 rev 2
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Initrd not found or empty - disabling initrd
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Zone ranges:
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Movable zone start for each node
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Early memory node ranges
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
Tue May 30 04:00:41 2023 kern.debug kernel: [    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
Tue May 30 04:00:41 2023 kern.debug kernel: [    0.000000] pcpu-alloc: [0] 0
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32480
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.000000] Kernel command line: console=ttyS0,115200 rootfstype=squashfs,jffs2
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Writing ErrCtl register=00000000
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Readback ErrCtl register=00000000
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] Memory: 119796K/131072K available (6562K kernel code, 585K rwdata, 1384K rodata, 1188K init, 214K bss, 11276K reserved, 0K cma-reserved)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] NR_IRQS: 51
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] CPU clock: 560.000 MHz
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 6825930166 ns
Tue May 30 04:00:41 2023 kern.info kernel: [    0.000002] sched_clock: 32 bits at 280MHz, resolution 3ns, wraps every 7669584382ns
Tue May 30 04:00:41 2023 kern.info kernel: [    0.008364] Calibrating delay loop... 278.93 BogoMIPS (lpj=1394688)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.084982] pid_max: default: 32768 minimum: 301
Tue May 30 04:00:41 2023 kern.info kernel: [    0.091055] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.098879] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.110404] cblist_init_generic: Setting adjustable number of callback queues.
Tue May 30 04:00:41 2023 kern.info kernel: [    0.118108] cblist_init_generic: Setting shift to 0 and lim to 1.
Tue May 30 04:00:41 2023 kern.info kernel: [    0.129539] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
Tue May 30 04:00:41 2023 kern.info kernel: [    0.140077] futex hash table entries: 256 (order: -1, 3072 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.147525] pinctrl core: initialized pinctrl subsystem
Tue May 30 04:00:41 2023 kern.info kernel: [    0.155072] NET: Registered PF_NETLINK/PF_ROUTE protocol family
Tue May 30 04:00:41 2023 kern.info kernel: [    0.183402] clocksource: Switched to clocksource MIPS
Tue May 30 04:00:41 2023 kern.info kernel: [    0.190634] NET: Registered PF_INET protocol family
Tue May 30 04:00:41 2023 kern.info kernel: [    0.196216] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.205018] tcp_listen_portaddr_hash hash table entries: 1024 (order: 0, 4096 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.214100] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.222353] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.230558] TCP bind hash table entries: 1024 (order: 1, 8192 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.238121] TCP: Hash tables configured (established 1024 bind 1024)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.245075] UDP hash table entries: 256 (order: 0, 4096 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.252071] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes, linear)
Tue May 30 04:00:41 2023 kern.info kernel: [    0.259980] NET: Registered PF_UNIX/PF_LOCAL protocol family
Tue May 30 04:00:41 2023 kern.info kernel: [    0.266115] PCI: CLS 0 bytes, default 32
Tue May 30 04:00:41 2023 kern.info kernel: [    0.274559] workingset: timestamp_bits=14 max_order=15 bucket_order=1
Tue May 30 04:00:41 2023 kern.info kernel: [    0.288053] squashfs: version 4.0 (2009/01/31) Phillip Lougher
Tue May 30 04:00:41 2023 kern.info kernel: [    0.294352] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
Tue May 30 04:00:41 2023 kern.info kernel: [    0.310360] pinctrl-single 1804002c.pinmux: 544 pins, size 68
Tue May 30 04:00:41 2023 kern.info kernel: [    0.317255] gpio-export gpio-export: 1 gpio(s) exported
Tue May 30 04:00:41 2023 kern.info kernel: [    0.324010] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
Tue May 30 04:00:41 2023 kern.info kernel: [    0.331760] printk: console [ttyS0] disabled
Tue May 30 04:00:41 2023 kern.info kernel: [    0.336477] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 11, base_baud = 2500000) is a 16550A
Tue May 30 04:00:41 2023 kern.info kernel: [    0.345809] printk: console [ttyS0] enabled
Tue May 30 04:00:41 2023 kern.info kernel: [    0.354729] printk: bootconsole [early0] disabled
Tue May 30 04:00:41 2023 kern.info kernel: [    0.382267] nand: device found, Manufacturer ID: 0x2c, Chip ID: 0xf1
Tue May 30 04:00:41 2023 kern.info kernel: [    0.388786] nand: Micron NAND 128MiB 3,3V 8-bit
Tue May 30 04:00:41 2023 kern.info kernel: [    0.393386] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
Tue May 30 04:00:41 2023 kern.info kernel: [    0.401153] Scanning device for bad blocks
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.490166] 8 fixed-partitions partitions found on MTD device ar934x-nand
Tue May 30 04:00:41 2023 kern.err kernel: [    0.497145] OF: Bad cell count for /ahb/nand@1b000200/partitions
Tue May 30 04:00:41 2023 kern.err kernel: [    0.503275] OF: Bad cell count for /ahb/nand@1b000200/partitions
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.509710] Creating 8 MTD partitions on "ar934x-nand":
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.515070] 0x000000000000-0x000000020000 : "u-boot"
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.523130] 0x000000020000-0x000000060000 : "caldata"
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.529676] 0x000000060000-0x0000000a0000 : "caldata-backup"
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.538731] 0x0000000a0000-0x000000120000 : "config"
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.545019] 0x000000120000-0x0000001a0000 : "pot"
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.553089] 0x0000001a0000-0x000007b00000 : "firmware"
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.560251] 2 fixed-partitions partitions found on MTD device firmware
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.566962] Creating 2 MTD partitions on "firmware":
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.572018] 0x000000000000-0x000000400000 : "kernel"
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.580388] 0x000000400000-0x000007960000 : "ubi"
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.587121] 0x000007b00000-0x000007d00000 : "language"
Tue May 30 04:00:41 2023 kern.notice kernel: [    0.595802] 0x000007d00000-0x000008000000 : "traffic_meter"
Tue May 30 04:00:41 2023 kern.info kernel: [    0.957140] switch0: Atheros AR8229 rev. 1 switch registered on mdio.0
Tue May 30 04:00:41 2023 kern.info kernel: [    1.434218] ag71xx 19000000.eth: connected to PHY at mdio.0:1f:00 [uid=004dd042, driver=Qualcomm Atheros QCA9561 built-in PHY]
Tue May 30 04:00:41 2023 kern.info kernel: [    1.446611] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
Tue May 30 04:00:41 2023 kern.info kernel: [    1.795549] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
Tue May 30 04:00:41 2023 kern.info kernel: [    1.805545] eth1: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
Tue May 30 04:00:41 2023 kern.info kernel: [    1.815575] NET: Registered PF_INET6 protocol family
Tue May 30 04:00:41 2023 kern.info kernel: [    1.832738] Segment Routing with IPv6
Tue May 30 04:00:41 2023 kern.info kernel: [    1.836637] In-situ OAM (IOAM) with IPv6
Tue May 30 04:00:41 2023 kern.info kernel: [    1.840780] NET: Registered PF_PACKET protocol family
Tue May 30 04:00:41 2023 kern.info kernel: [    1.846045] 8021q: 802.1Q VLAN Support v1.8
Tue May 30 04:00:41 2023 kern.info kernel: [    1.859104] PCI host bridge to bus 0000:00
Tue May 30 04:00:41 2023 kern.info kernel: [    1.863287] pci_bus 0000:00: root bus resource [mem 0x10000000-0x13ffffff]
Tue May 30 04:00:41 2023 kern.info kernel: [    1.870341] pci_bus 0000:00: root bus resource [io  0x0000]
Tue May 30 04:00:41 2023 kern.info kernel: [    1.876025] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
Tue May 30 04:00:41 2023 kern.info kernel: [    1.884136] pci 0000:00:00.0: [168c:003c] type 00 class 0x028000
Tue May 30 04:00:41 2023 kern.info kernel: [    1.890275] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x001fffff 64bit]
Tue May 30 04:00:41 2023 kern.info kernel: [    1.897237] pci 0000:00:00.0: reg 0x30: [mem 0x00000000-0x0000ffff pref]
Tue May 30 04:00:41 2023 kern.info kernel: [    1.904147] pci 0000:00:00.0: supports D1 D2
Tue May 30 04:00:41 2023 kern.info kernel: [    1.910024] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
Tue May 30 04:00:41 2023 kern.info kernel: [    1.916837] pci 0000:00:00.0: BAR 0: assigned [mem 0x10000000-0x101fffff 64bit]
Tue May 30 04:00:41 2023 kern.info kernel: [    1.924315] pci 0000:00:00.0: BAR 6: assigned [mem 0x10200000-0x1020ffff pref]
Tue May 30 04:00:41 2023 kern.notice kernel: [    1.937455] UBI: auto-attach mtd7
Tue May 30 04:00:41 2023 kern.notice kernel: [    1.940874] ubi0: attaching mtd7
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.891112] ubi0: scanning is finished
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.906718] ubi0: attached mtd7 (name "ubi", size 117 MiB)
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.912319] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.919361] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.926272] ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.933338] ubi0: good PEBs: 939, bad PEBs: 0, corrupted PEBs: 0
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.939453] ubi0: user volume: 2, internal volumes: 1, max. volumes count: 128
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.946800] ubi0: max/mean erase counter: 4/1, WL threshold: 4096, image sequence number: 2887050566
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.956094] ubi0: available PEBs: 0, total reserved PEBs: 939, PEBs reserved for bad PEB handling: 20
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.965525] ubi0: background thread "ubi_bgt0d" started, PID 213
Tue May 30 04:00:41 2023 kern.info kernel: [    2.973603] block ubiblock0_0: created from ubi0:0(rootfs)
Tue May 30 04:00:41 2023 kern.notice kernel: [    2.979200] ubiblock: device ubiblock0_0 (rootfs) set to be root filesystem
Tue May 30 04:00:41 2023 kern.info kernel: [    2.993467] VFS: Mounted root (squashfs filesystem) readonly on device 254:0.
Tue May 30 04:00:41 2023 kern.info kernel: [    3.006346] Freeing unused kernel image (initmem) memory: 1188K
Tue May 30 04:00:41 2023 kern.warn kernel: [    3.012366] This architecture does not have kernel memory protection.
Tue May 30 04:00:41 2023 kern.info kernel: [    3.018941] Run /sbin/init as init process
Tue May 30 04:00:41 2023 kern.debug kernel: [    3.023100]   with arguments:
Tue May 30 04:00:41 2023 kern.debug kernel: [    3.023106]     /sbin/init
Tue May 30 04:00:41 2023 kern.debug kernel: [    3.023113]   with environment:
Tue May 30 04:00:41 2023 kern.debug kernel: [    3.023119]     HOME=/
Tue May 30 04:00:41 2023 kern.debug kernel: [    3.023125]     TERM=linux
Tue May 30 04:00:41 2023 user.info kernel: [    3.551446] init: Console is alive
Tue May 30 04:00:41 2023 user.info kernel: [    3.555544] init: - watchdog -
Tue May 30 04:00:41 2023 user.info kernel: [    4.805749] kmodloader: loading kernel modules from /etc/modules-boot.d/*
Tue May 30 04:00:41 2023 kern.info kernel: [    4.877504] usbcore: registered new interface driver usbfs
Tue May 30 04:00:41 2023 kern.info kernel: [    4.883186] usbcore: registered new interface driver hub
Tue May 30 04:00:41 2023 kern.info kernel: [    4.888727] usbcore: registered new device driver usb
Tue May 30 04:00:41 2023 kern.info kernel: [    4.915247] ehci-platform 1b000000.usb: EHCI Host Controller
Tue May 30 04:00:41 2023 kern.info kernel: [    4.921043] ehci-platform 1b000000.usb: new USB bus registered, assigned bus number 1
Tue May 30 04:00:41 2023 kern.info kernel: [    4.929182] ehci-platform 1b000000.usb: irq 3, io mem 0x1b000000
Tue May 30 04:00:41 2023 kern.info kernel: [    4.963429] ehci-platform 1b000000.usb: USB 2.0 started, EHCI 1.00
Tue May 30 04:00:41 2023 kern.info kernel: [    4.970786] hub 1-0:1.0: USB hub found
Tue May 30 04:00:41 2023 kern.info kernel: [    4.975589] hub 1-0:1.0: 1 port detected
Tue May 30 04:00:41 2023 user.info kernel: [    4.983134] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
Tue May 30 04:00:41 2023 user.info kernel: [    5.001636] init: - preinit -
Tue May 30 04:00:41 2023 kern.notice kernel: [    7.063432] random: crng init done
Tue May 30 04:00:41 2023 kern.info kernel: [    8.213821] eth1: link up (1000Mbps/Full duplex)
Tue May 30 04:00:41 2023 kern.info kernel: [    8.223500] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
Tue May 30 04:00:41 2023 kern.info kernel: [    8.230330] IPv6: ADDRCONF(NETDEV_CHANGE): eth1.1: link becomes ready
Tue May 30 04:00:41 2023 kern.notice kernel: [   10.503800] UBIFS (ubi0:1): Mounting in unauthenticated mode
Tue May 30 04:00:41 2023 kern.notice kernel: [   10.509797] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" started, PID 353
Tue May 30 04:00:41 2023 kern.notice kernel: [   10.605689] UBIFS (ubi0:1): recovery needed
Tue May 30 04:00:41 2023 kern.notice kernel: [   10.767184] UBIFS (ubi0:1): recovery completed
Tue May 30 04:00:41 2023 kern.notice kernel: [   10.771832] UBIFS (ubi0:1): UBIFS: mounted UBI device 0, volume 1, name "rootfs_data"
Tue May 30 04:00:41 2023 kern.notice kernel: [   10.779849] UBIFS (ubi0:1): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
Tue May 30 04:00:41 2023 kern.notice kernel: [   10.789936] UBIFS (ubi0:1): FS size: 110723072 bytes (105 MiB, 872 LEBs), max 883 LEBs, journal size 5586944 bytes (5 MiB, 44 LEBs)
Tue May 30 04:00:41 2023 kern.notice kernel: [   10.801964] UBIFS (ubi0:1): reserved for root: 4952683 bytes (4836 KiB)
Tue May 30 04:00:41 2023 kern.notice kernel: [   10.808695] UBIFS (ubi0:1): media format: w5/r0 (latest is w5/r0), UUID 41517540-7EE2-46D6-8378-CF4AAA389A2E, small LPT model
Tue May 30 04:00:41 2023 user.info kernel: [   10.823961] mount_root: switching to ubifs overlay
Tue May 30 04:00:41 2023 user.warn kernel: [   10.841753] urandom-seed: Seeding with /etc/urandom.seed
Tue May 30 04:00:41 2023 kern.info kernel: [   10.932376] eth1: link down
Tue May 30 04:00:41 2023 user.info kernel: [   10.960521] procd: - early -
Tue May 30 04:00:41 2023 user.info kernel: [   10.963955] procd: - watchdog -
Tue May 30 04:00:41 2023 user.info kernel: [   11.693138] procd: - watchdog -
Tue May 30 04:00:41 2023 user.info kernel: [   11.697491] procd: - ubus -
Tue May 30 04:00:41 2023 user.info kernel: [   11.778590] procd: - init -
Tue May 30 04:00:41 2023 user.info kernel: [   13.104318] kmodloader: loading kernel modules from /etc/modules.d/*
Tue May 30 04:00:41 2023 user.info kernel: [   13.614545] urngd: v1.0.2 started.
Tue May 30 04:00:41 2023 kern.info kernel: [   13.824096] Loading modules backported from Linux version v6.1.24-0-g0102425ac76b
Tue May 30 04:00:41 2023 kern.info kernel: [   13.831713] Backport generated by backports.git v5.15.92-1-44-gd6ea70fafd36
Tue May 30 04:00:41 2023 kern.info kernel: [   14.282981] PPP generic driver version 2.4.2
Tue May 30 04:00:41 2023 kern.info kernel: [   14.290990] NET: Registered PF_PPPOX protocol family
Tue May 30 04:00:41 2023 kern.err kernel: [   14.337118] ath10k_pci 0000:00:00.0: of_irq_parse_pci: failed with rc=-14
Tue May 30 04:00:41 2023 kern.info kernel: [   14.344102] ath10k 5.12 driver, optimized for CT firmware, probing pci device: 0x3c.
Tue May 30 04:00:41 2023 kern.info kernel: [   14.354318] ath10k_pci 0000:00:00.0: enabling device (0000 -> 0002)
Tue May 30 04:00:41 2023 kern.info kernel: [   14.360886] ath10k_pci 0000:00:00.0: pci irq legacy oper_irq_mode 1 irq_mode 0 reset_mode 0
Tue May 30 04:00:41 2023 kern.info kernel: [   18.050486] ath10k_pci 0000:00:00.0: qca988x hw2.0 target 0x4100016c chip_id 0x043222ff sub 0000:0000
Tue May 30 04:00:41 2023 kern.info kernel: [   18.059936] ath10k_pci 0000:00:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
Tue May 30 04:00:41 2023 kern.info kernel: [   18.072961] ath10k_pci 0000:00:00.0: firmware ver 10.1-ct-8x-__fW-022-ecad3248 api 2 features wmi-10.x,has-wmi-mgmt-tx,mfp,txstatus-noack,wmi-10.x-CT,ratemask-CT,txrate-CT,get-temp-CT,tx-rc-CT,cust-stats-CT,retry-gt2-CT,txrate2-CT,beacon-cb-CT,wmi-block-ack-CT crc32 3e4cf97f
Tue May 30 04:00:41 2023 kern.info kernel: [   19.045378] ath10k_pci 0000:00:00.0: board_file api 1 bmi_id N/A crc32 bebc7c08
Tue May 30 04:00:41 2023 kern.warn kernel: [   21.063413] ath10k_pci 0000:00:00.0: failed to receive control response completion, polling..
Tue May 30 04:00:41 2023 kern.err kernel: [   22.103417] ath10k_pci 0000:00:00.0: Service connect timeout
Tue May 30 04:00:41 2023 kern.err kernel: [   22.109188] ath10k_pci 0000:00:00.0: failed to connect htt (-145)
Tue May 30 04:00:41 2023 kern.err kernel: [   22.246467] ath10k_pci 0000:00:00.0: could not init core (-145)
Tue May 30 04:00:41 2023 kern.err kernel: [   22.252599] ath10k_pci 0000:00:00.0: could not probe fw (-145)
Tue May 30 04:00:41 2023 kern.debug kernel: [   22.312437] ath: EEPROM regdomain: 0x0
Tue May 30 04:00:41 2023 kern.debug kernel: [   22.312468] ath: EEPROM indicates default country code should be used
Tue May 30 04:00:41 2023 kern.debug kernel: [   22.312476] ath: doing EEPROM country->regdmn map search
Tue May 30 04:00:41 2023 kern.debug kernel: [   22.312494] ath: country maps to regdmn code: 0x3a
Tue May 30 04:00:41 2023 kern.debug kernel: [   22.312504] ath: Country alpha2 being used: US
Tue May 30 04:00:41 2023 kern.debug kernel: [   22.312513] ath: Regpair used: 0x3a
Tue May 30 04:00:41 2023 kern.debug kernel: [   22.326136] ieee80211 phy1: Selected rate control algorithm 'minstrel_ht'
Tue May 30 04:00:41 2023 kern.info kernel: [   22.328451] ieee80211 phy1: Atheros AR9340 Rev:2 mem=0xb8100000, irq=15
Tue May 30 04:00:41 2023 user.info kernel: [   22.354004] kmodloader: done loading kernel modules from /etc/modules.d/*
Tue May 30 04:00:43 2023 user.notice dnsmasq: DNS rebinding protection is active, will discard upstream RFC1918 responses!
Tue May 30 04:00:43 2023 user.notice dnsmasq: Allowing 127.0.0.0/8 responses
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: started, version 2.89 cachesize 1000
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: DNS service limited to local subnets
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: compile time options: IPv6 GNU-getopt no-DBus UBus no-i18n no-IDN DHCP no-DHCPv6 no-Lua TFTP no-conntrack no-ipset no-nftset no-auth no-cryptohash no-DNSSEC no-ID loop-detect inotify dumpfile
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: UBus support enabled: connected to system bus
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: using only locally-known addresses for test
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: using only locally-known addresses for onion
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: using only locally-known addresses for localhost
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: using only locally-known addresses for local
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: using only locally-known addresses for invalid
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: using only locally-known addresses for bind
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: using only locally-known addresses for lan
Tue May 30 04:00:44 2023 daemon.warn dnsmasq[1]: no servers found in /tmp/resolv.conf.d/resolv.conf.auto, will retry
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: read /etc/hosts - 12 names
Tue May 30 04:00:44 2023 daemon.info dnsmasq[1]: read /tmp/hosts/dhcp.cfg01411c - 0 names
Tue May 30 04:00:45 2023 authpriv.info dropbear[1212]: Not backgrounding
Tue May 30 04:00:49 2023 daemon.notice wpa_supplicant[1321]: Successfully initialized wpa_supplicant
Tue May 30 04:00:50 2023 user.notice : Added device handler type: bonding
Tue May 30 04:00:50 2023 user.notice : Added device handler type: 8021ad
Tue May 30 04:00:50 2023 user.notice : Added device handler type: 8021q
Tue May 30 04:00:50 2023 user.notice : Added device handler type: macvlan
Tue May 30 04:00:50 2023 user.notice : Added device handler type: veth
Tue May 30 04:00:50 2023 user.notice : Added device handler type: bridge
Tue May 30 04:00:50 2023 user.notice : Added device handler type: Network device
Tue May 30 04:00:50 2023 user.notice : Added device handler type: tunnel
Tue May 30 04:00:53 2023 daemon.notice procd: /etc/rc.d/S50uhttpd: 4+0 records in
Tue May 30 04:00:53 2023 daemon.notice procd: /etc/rc.d/S50uhttpd: 4+0 records out
Tue May 30 04:00:54 2023 user.notice ucitrack: Setting up /etc/config/dhcp reload dependency on /etc/config/network
Tue May 30 04:00:54 2023 user.notice ucitrack: Setting up /etc/config/network reload dependency on /etc/config/wireless
Tue May 30 04:00:54 2023 user.notice ucitrack: Setting up /etc/config/luci-splash reload dependency on /etc/config/firewall
Tue May 30 04:00:55 2023 user.notice ucitrack: Setting up /etc/config/qos reload dependency on /etc/config/firewall
Tue May 30 04:00:55 2023 user.notice ucitrack: Setting up /etc/config/miniupnpd reload dependency on /etc/config/firewall
Tue May 30 04:00:55 2023 user.notice ucitrack: Setting up /etc/config/odhcpd reload dependency on /etc/config/dhcp
Tue May 30 04:00:56 2023 user.notice ucitrack: Setting up non-init /etc/config/fstab reload handler: /sbin/block mount
Tue May 30 04:00:56 2023 user.notice ucitrack: Setting up /etc/config/system reload trigger for non-procd /etc/init.d/led
Tue May 30 04:00:57 2023 user.notice ucitrack: Setting up /etc/config/luci_statistics reload dependency on /etc/config/system
Tue May 30 04:00:57 2023 user.notice ucitrack: Setting up /etc/config/dhcp reload dependency on /etc/config/system
Tue May 30 04:00:58 2023 daemon.notice procd: /etc/rc.d/S96led: setting up led WAN:ACT
Tue May 30 04:00:58 2023 daemon.notice procd: /etc/rc.d/S96led: setting up led WAN:LINK
Tue May 30 04:00:59 2023 kern.info kernel: [   43.037069] eth1: link up (1000Mbps/Full duplex)
Tue May 30 04:00:59 2023 kern.info kernel: [   43.041808] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
Tue May 30 04:00:59 2023 kern.info kernel: [   43.064074] br-lan: port 1(eth1.1) entered blocking state
Tue May 30 04:00:59 2023 kern.info kernel: [   43.069592] br-lan: port 1(eth1.1) entered disabled state
Tue May 30 04:00:59 2023 kern.info kernel: [   43.075509] device eth1.1 entered promiscuous mode
Tue May 30 04:00:59 2023 kern.info kernel: [   43.080396] device eth1 entered promiscuous mode
Tue May 30 04:00:59 2023 kern.info kernel: [   43.121364] br-lan: port 1(eth1.1) entered blocking state
Tue May 30 04:00:59 2023 kern.info kernel: [   43.126935] br-lan: port 1(eth1.1) entered forwarding state
Tue May 30 04:00:59 2023 daemon.notice netifd: Interface 'lan' is enabled
Tue May 30 04:00:59 2023 daemon.notice netifd: Interface 'lan' is setting up now
Tue May 30 04:00:59 2023 daemon.notice netifd: Interface 'lan' is now up
Tue May 30 04:00:59 2023 daemon.notice netifd: bridge 'br-lan' link is up
Tue May 30 04:01:00 2023 daemon.notice netifd: Interface 'lan' has link connectivity
Tue May 30 04:01:00 2023 daemon.notice netifd: VLAN 'eth1.1' link is up
Tue May 30 04:01:00 2023 daemon.notice netifd: Interface 'loopback' is enabled
Tue May 30 04:01:00 2023 daemon.notice netifd: Interface 'loopback' is setting up now
Tue May 30 04:01:00 2023 daemon.notice netifd: Interface 'loopback' is now up
Tue May 30 04:01:00 2023 daemon.notice netifd: Interface 'wan' is enabled
Tue May 30 04:01:00 2023 daemon.notice netifd: Interface 'wan6' is enabled
Tue May 30 04:01:00 2023 daemon.notice netifd: Network device 'eth1' link is up
Tue May 30 04:01:00 2023 daemon.notice netifd: Network device 'lo' link is up
Tue May 30 04:01:00 2023 daemon.notice netifd: Interface 'loopback' has link connectivity
Tue May 30 04:01:00 2023 kern.info kernel: [   44.103830] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
Tue May 30 04:01:02 2023 daemon.info procd: - init complete -
Tue May 30 04:01:02 2023 user.notice firewall: Reloading firewall due to ifup of lan (br-lan)
Tue May 30 04:01:04 2023 daemon.notice hostapd: Configuration file: /var/run/hostapd-phy1.conf (phy phy1-ap0) --> new PHY
Tue May 30 04:01:04 2023 kern.info kernel: [   48.008435] br-lan: port 2(phy1-ap0) entered blocking state
Tue May 30 04:01:04 2023 kern.info kernel: [   48.014285] br-lan: port 2(phy1-ap0) entered disabled state
Tue May 30 04:01:04 2023 kern.info kernel: [   48.020378] device phy1-ap0 entered promiscuous mode
Tue May 30 04:01:04 2023 kern.info kernel: [   48.079137] IPv6: ADDRCONF(NETDEV_CHANGE): phy1-ap0: link becomes ready
Tue May 30 04:01:04 2023 kern.info kernel: [   48.086259] br-lan: port 2(phy1-ap0) entered blocking state
Tue May 30 04:01:04 2023 kern.info kernel: [   48.091966] br-lan: port 2(phy1-ap0) entered forwarding state
Tue May 30 04:01:04 2023 daemon.notice hostapd: phy1-ap0: interface state UNINITIALIZED->ENABLED
Tue May 30 04:01:04 2023 daemon.notice hostapd: phy1-ap0: AP-ENABLED
Tue May 30 04:01:06 2023 daemon.notice netifd: Wireless device 'radio0' is now up
Tue May 30 04:01:06 2023 daemon.notice netifd: Network device 'phy1-ap0' link is up
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: started, version 2.89 cachesize 1000
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: DNS service limited to local subnets
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: compile time options: IPv6 GNU-getopt no-DBus UBus no-i18n no-IDN DHCP no-DHCPv6 no-Lua TFTP no-conntrack no-ipset no-nftset no-auth no-cryptohash no-DNSSEC no-ID loop-detect inotify dumpfile
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: UBus support enabled: connected to system bus
Tue May 30 04:01:08 2023 daemon.info dnsmasq-dhcp[1]: DHCP, IP range 192.168.1.100 -- 192.168.1.249, lease time 12h
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: using only locally-known addresses for test
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: using only locally-known addresses for onion
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: using only locally-known addresses for localhost
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: using only locally-known addresses for local
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: using only locally-known addresses for invalid
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: using only locally-known addresses for bind
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: using only locally-known addresses for lan
Tue May 30 04:01:08 2023 daemon.warn dnsmasq[1]: no servers found in /tmp/resolv.conf.d/resolv.conf.auto, will retry
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: read /etc/hosts - 12 names
Tue May 30 04:01:08 2023 daemon.info dnsmasq[1]: read /tmp/hosts/dhcp.cfg01411c - 4 names
Tue May 30 04:01:08 2023 daemon.info dnsmasq-dhcp[1]: read /etc/ethers - 0 addresses
Tue May 30 04:05:15 2023 daemon.info hostapd: phy1-ap0: STA 48:e7:da:f5:4d:37 IEEE 802.11: authenticated
Tue May 30 04:05:15 2023 daemon.info hostapd: phy1-ap0: STA 48:e7:da:f5:4d:37 IEEE 802.11: associated (aid 1)
Tue May 30 04:05:15 2023 daemon.notice hostapd: phy1-ap0: AP-STA-CONNECTED 48:e7:da:f5:4d:37 auth_alg=open
Tue May 30 04:05:16 2023 daemon.info dnsmasq-dhcp[1]: DHCPREQUEST(br-lan) 192.168.1.104 48:e7:da:f5:4d:37
Tue May 30 04:05:16 2023 daemon.info dnsmasq-dhcp[1]: DHCPACK(br-lan) 192.168.1.104 48:e7:da:f5:4d:37 Aspire-V-Nitro
Tue May 30 04:05:22 2023 daemon.err uhttpd[1617]: [info] luci: accepted login on / for root from 192.168.1.104
```
</details>

DragonBluep committed:
ath79: copy patches and kernel config from 5.15 to 6.1

This is preparation for kernel 6.1 support.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ath79: fix ethernet driver build errors on kernel 6.1

Some net APIs have changed on the new kernel. Update them to fix
compile errors.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ath79: backport gpio immutable irq_chip support

This patch converts the driver to immutable irq-chip, which can
silence some gpio warnings.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

guidosarducci commented:
Thanks for this, I'll try building for WZR-HP-G300NH (ath79/generic) when I have a chance.

One thing I noticed after a quick look: why are you modifying the 5.15 kernel config given this is a 6.1 PR?

DragonBluep commented:
> Thanks for this, I'll try building for WZR-HP-G300NH (ath79/generic) when I have a chance.
> 
> One thing I noticed after a quick look: why are you modifying the 5.15 kernel config given this is a 6.1 PR?

ath79 target has some common config files such as `target/linux/ath79/generic/config-default` used for both kernel 5.15 and 6.1. I have to refresh `config-5.15` to make sure kernel 5.15 still work with these changed configs.

guidosarducci commented:
> ath79 target has some common config files such as `target/linux/ath79/generic/config-default` used for both kernel 5.15 and 6.1. I have to refresh `config-5.15` to make sure kernel 5.15 still work with these changed configs.

OK, I see. Probably good to note that for the commit history and to avoid confusion.

I see build is missing a generic kernel setting with my config (looks like this was added in kernel 6.1):
```
  sysctl test driver (TEST_SYSCTL) [N/m/y/?] n
  udelay test driver (TEST_UDELAY) [N/m/y/?] n
  Test static keys (TEST_STATIC_KEYS) [N/m/?] n
  Test DYNAMIC_DEBUG (TEST_DYNAMIC_DEBUG) [N/m/y/?] (NEW) make[7]: *** [scripts/kconfig/Makefile:77: syncconfig] Error 1
make[6]: *** [Makefile:704: syncconfig] Error 2
```
Fix:
```
diff --git a/target/linux/generic/config-6.1 b/target/linux/generic/config-6.1
index 94c97f9f54..c94ad1e18d 100644
--- a/target/linux/generic/config-6.1
+++ b/target/linux/generic/config-6.1
@@ -6675,2 +6675,3 @@ CONFIG_TCP_CONG_CUBIC=y
 # CONFIG_TEST_DIV64 is not set
+# CONFIG_TEST_DYNAMIC_DEBUG is not set
 # CONFIG_TEST_FIRMWARE is not set
```

With that the build succeeds, but I've yet to flash and run. Will follow up later...

DragonBluep committed:
ath79: add missing symbols by refreshing kernel configs

Some symbols are outdated or missing due to daily kernel bumps. It's
better to re-add them. All configs are automatically refreshed by
'make kernel_oldconfig CONFIG_TARGET=taget' and
'make kernel_oldconfig CONFIG_TARGET=subtarget'

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

mrkiko commented:
Very nice you're using the R6100 :)

DragonBluep commented:
v2:
1. Refresh kernel configs for 5.15 before port to 6.1 so that we can clearly see which kernel symbol changed.
2. Correct typo 'taregt'.
3. Fix build error on ath5k.

@guidosarducci I refreshed them again but still not see this new symbol. Have you enabled some additional features?

robimarko commented:
That looks more like a PCI issue rather than ath10k-ct itself

DragonBluep committed:
ath79: enable variable sector size erasure for generic subtarget

Make use of minor sector size (4k) erasure on supported flash chips
to improve spi read/write performance.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

Ansuel commented:
sorry but the 200 patch looks totally wrong and the real reason should be found.... there is probably and error in how sizeof is calculated

chunkeey commented:
> sorry but the 200 patch looks totally wrong and the real reason should be found.... there is probably and error in how sizeof is calculated

ran into this as well.  Yes, this looks suspicious. I think this is a compiler bug. Because it doesn't complain about the 
memset in the same function ath9k_ps_wakeup() just the line above.

| memset(&common->cc_survey, 0, sizeof(common->cc_survey));

of note here is that cc_survey is of the same struct ath_cycle_counters! 

Now the change to struct_group in ath_cycle_counters by itself isn't "wrong". If I was @DragonBluep then I would just post it to linux-wireless mailing list.  They could be somewhat inclined to accept it (maybe just hint that this struct_group conversion is going on at the moment). What do you think?

---
a less intrusive way to do it: The error/warning will go away too, though it really shows how wonky fortify_source heuristic sometimes is.

```
--- a/drivers/net/wireless/ath/ath9k/main.c        2023-05-28 22:09:33.069932579 +0200
+++ b/drivers/net/wireless/ath/ath9k/main.c     2023-05-28 22:09:26.237726389 +0200
@@ -137,7 +137,7 @@ void ath9k_ps_wakeup(struct ath_softc *s
                spin_lock(&common->cc_lock);
                ath_hw_cycle_counters_update(common);
                memset(&common->cc_survey, 0, sizeof(common->cc_survey));
-               memset(&common->cc_ani, 0, sizeof(common->cc_ani));
+               common->cc_ani.cycles = common->cc_ani.rx_busy = common->cc_ani.rx_frame = common->cc_ani.tx_frame;
                spin_unlock(&common->cc_lock);
        }
```

Ansuel commented:
@chunkeey we are totally missing something or it's a bug in the fortify scan... for sure it doesn't make sense to do struct_group on the entire struct since you should be able to just reference the struct directly since the size should be the same... I'm very confused by the error

chunkeey commented:
> @chunkeey we are totally missing something or it's a bug in the fortify scan... for sure it doesn't make sense to do struct_group on the entire struct since you should be able to just reference the struct directly since the size should be the same... I'm very confused by the error

yes precisely, this doesn't make sense. I don't know if you have looked it up. The thing in ath9k_ps_wakeup is that the code is:

    memset(&common->cc_survey, 0, sizeof(common->cc_survey));
    memset(&common->cc_ani, 0, sizeof(common->cc_ani));

these two lines are right next to one another. But the error happens only with the second memset. However both cc_ani and cc_survey are of the same ath_cycle_counters struct! That's what I'm trying to spotlight. The error happens with cc_ani memset, but it should also happen with cc_survey memset too! 

This wonky behavior is why I'm recommending posting this patch upstream too. I'm sure this could trigger some watchful eyes there and we get a resolution this way.

guidosarducci commented:
> v2:
> 
>     1. Refresh kernel configs for 5.15 before port to 6.1 so that we can clearly see which kernel symbol changed.
> 
>     2. Correct typo 'taregt'.
> 
>     3. Fix build error on ath5k.
> 
> 
> @guidosarducci I refreshed them again but still not see this new symbol. Have you enabled some additional features?

Thanks for those changes. Note that the existing OpenWrt config option `CONFIG_KERNEL_DYNAMIC_DEBUG=y` is what ends up exposing the new kernel 6.1 setting `# CONFIG_TEST_DYNAMIC_DEBUG is not set`. This is required to support 6.1, so if you could add that here then someone else won't have to do it later or perhaps ends up forgotten. Thanks!

DragonBluep committed:
ath79: ignore the abused interrupt-map on PCIe node

ath79 PCIe interrupt controller has stopped working correctly. This
is because the DT exposing a non-sensical interrupt-map property,
and their drivers relying on the kernel ignoring this property[1].

This patch fixes the PCIe init error:
ath9k 0000:00:00.0: of_irq_parse_pci: failed with rc=-14

Notice:
This is just a workaround, not a fix. PCIe driver and related dts
node need to be rewritten.

[1] https://lore.kernel.org/all/20211201114102.13446-1-maz@kernel.org/

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ath79: rename and sort patches by OpenWrt naming rules

The patches in the ath79 target have not been sorted for a long time
and they are very chaotic now. This patch sorts them again according
to the OpenWrt naming rules[1], so that we can better manage them.

[1] https://openwrt.org/docs/guide-developer/toolchain/use-patches-with-buildsystem#naming_patches

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ath79: refresh patches and configs to introduce kernel 6.1 support

All kernel configs are refreshed by
'make kernel_oldconfig CONFIG_TARGET=target' and
'make kernel_oldconfig CONFIG_TARGET=subtarget'.

upstreamed patches:
010-v5.17-spi-ar934x-fix-transfer-and-word-delays.patch
011-v5.17-spi-ar934x-fix-transfer-size.patch
020-v5.18-spi-ath79-Implement-the-spi_mem-interface.patch
030-v5.18-ath79-add-support-for-booting-QCN550x.patch

build and run tested on:
ath79/generic/ar7241
ath79/generic/qca9563
ath79/nand/ar9344

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep commented:
v3:
1. Fix pcie init issue (which misunderstood as ath10k-ct issue)
2. Enable variable sector size erasure for generic subtarget
3. Rename and sort ath79 target patches by OpenWrt naming rules.

DragonBluep commented:
> ```
> memset(&common->cc_survey, 0, sizeof(common->cc_survey));
> memset(&common->cc_ani, 0, sizeof(common->cc_ani));
> ```
> 
> 
>     
>   
> 
> these two lines are right next to one another. But the error happens only with the second memset. However both cc_ani and cc_survey are of the same ath_cycle_counters struct! That's what I'm trying to spotlight. The error happens with cc_ani memset, but it should also happen with cc_survey memset too!
> 
> 
> This wonky behavior is why I'm recommending posting this patch upstream too. I'm sure this could trigger some watchful eyes there and we get a resolution this way.

I even tried moving `memset(&common->cc_ani, 0, sizeof(common->cc_ani));` to the front of `memset(&common->cc_survey, 0, sizeof(common->cc_survey));`, however the compiler still complained about `memset(&common->cc_ani, 0, sizeof(common->cc_ani));`. I have also tried replacing `sizeof(common->cc_ani)` with `16` even `8`, but it still doesn't work.

I will try to send this patch upstream. Although I am pretty sure that it will not be accepted since it is not a real fix. 😂

DragonBluep commented:
> Note that the existing OpenWrt config option `CONFIG_KERNEL_DYNAMIC_DEBUG=y` is what ends up exposing the new kernel 6.1 setting `# CONFIG_TEST_DYNAMIC_DEBUG is not set`. This is required to support 6.1, so if you could add that here then someone else won't have to do it later or perhaps ends up forgotten. Thanks!

@guidosarducci Thanks for your report. It should be a build system issue. Maybe we need to add a new symbol to https://github.com/openwrt/openwrt/blob/main/config/Config-kernel.in. However I think it goes beyond the topic of this PR. And this problem should affect **all target** so we'd better to create a new issue to discuss it and open a new PR to fix it.

DragonBluep commented:
> sorry but the 200 patch looks totally wrong and the real reason should be found.... there is probably and error in how sizeof is calculated

I agree that this workaround is terrible, and that is why I don't call it fix. `sizeof()` should be okay since `cnts` should has the same size as the entire structure.

guidosarducci commented:
> > Note that the existing OpenWrt config option `CONFIG_KERNEL_DYNAMIC_DEBUG=y` is what ends up exposing the new kernel 6.1 setting `# CONFIG_TEST_DYNAMIC_DEBUG is not set`. This is required to support 6.1, so if you could add that here then someone else won't have to do it later or perhaps ends up forgotten. Thanks!
> 
> @guidosarducci Thanks for your report. It should be a build system issue. Maybe we need to add a new symbol to https://github.com/openwrt/openwrt/blob/main/config/Config-kernel.in. 

No, not a build system issue or something requiring Config-kernel.in changes. It's normal that if upstream Kconfig rules change we may need to preset additional generic config options.

> However I think it goes beyond the topic of this PR. And this problem should affect **all target** so we'd better to create a new issue to discuss it and open a new PR to fix it.

No problem, if you prefer we'll fix it elsewhere.

timocapa commented:
FritzBox 7490 (WASP) compiles & boots. WiFi (ath10k-ct (QCA9880) and ath9k (QCA9550)) works. Using GCC 13.

robimarko commented:
This ath9k warning, it occurs with GCC12 or 13?

Cause, on ipq807x with GCC12 I am not hitting it at all

DragonBluep commented:
> This ath9k warning, it occurs with GCC12 or 13?

@robimarko I am using the default GCC 12. Let's see if CI will complain about it.

robimarko commented:
Hm, then its either MIPS or target related as on ipq807x which is ARM64/A53 I get no warnings with GCC12

DragonBluep commented:
Resolve conflict, refresh patches && kernel configs, nothing new.

DragonBluep commented:
@chunkeey @Ansuel @PolynomialDivision GCC 13 seems to have fixed the noisy `__write_overflow_field` compiler issue. https://github.com/openwrt/openwrt/commit/6fe01c2994d4439c13703452a6e28daf79de6714

DragonBluep commented:
I'd like to say this PR is ready.

DragonBluep commented:
Update: Resolve conflicts.

timocapa commented:
Been running this MR since I commented on it on May 31 and I currently have over 35 days of uptime with no issues

blocktrron commented:
Merged to my staging tree, thanks a lot!
